tests/benchmark/fetch-download.js

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+
+import url from "node:url";
+import { Worker } from "node:worker_threads";
+
+const __dirname = url.fileURLToPath(new URL(".", import.meta.url));
+
+const USE_VIRTIO = !!process.env.USE_VIRTIO;
+const BENCHFILE_SIZE = (parseInt(process.env.BENCHFILE_SIZE_MB, 10) || 32) * 1024 * 1024;
+
+const { V86 } = await import("../../build/libv86.mjs");
+
+const LOG_SERIAL = true;
+const SHOW_LOGS = false;
+
+var SERVER_PORT = parseInt(process.env.SERVER_PORT, 10) || 0;
+var server = null;
+
+const emulator = new V86({
+    bios: { url: __dirname + "/../../bios/seabios.bin" },
+    vga_bios: { url: __dirname + "/../../bios/vgabios.bin" },
+    bzimage: { url: __dirname + "/../../images/buildroot-bzimage68.bin" },
+    autostart: true,
+    memory_size: 64 * 1024 * 1024,
+    net_device: {
+        relay_url: "fetch",
+        type: USE_VIRTIO ? "virtio" : "ne2k",
+        local_http: true
+    }
+});
+
+emulator.bus.register("emulator-ready", function()
+{
+    server = new Worker(__dirname + "../devices/fetch_testserver.js", { workerData: { port: SERVER_PORT, benchsize: BENCHFILE_SIZE } });
```

Comment:
I think this code would be overall more readable if you inline the worker code in here (by creating a function and converting to string, if possible: https://github.com/copy/v86/blob/a3840e32b30d7005f2e7533a557b769333127425/src/main.js#L135)

Then simplify the worker (inline get_handler, only handle `/bench`).

Same for the test case.

Reply:
Worker threads only supports script files, so converting code to string is not suitable here.

```
> the_worker = () => console.log("hello")

> url = URL.createObjectURL(new Blob(["(" + the_worker.toString() + ")()"], { type: "text/javascript" }))

> const { Worker } = await import("node:worker_threads")

> new Worker(url)
Uncaught:
TypeError [ERR_WORKER_PATH]: The worker script or module filename must be an absolute path or a relative path starting with './' or '../'. Received "blob:nodedata:<...>"
    at new Worker (node:internal/worker:187:15) {
  code: 'ERR_WORKER_PATH'
}
```